lark/tree_templates.py

```diff
@@ -29,24 +30,26 @@ def test_var(self, var: Union[Tree, str]) -> Optional[str]:
             var: Tree | str - The tree node to test
 
         """
-        if isinstance(var, str) and var.startswith('$'):
-            return var.lstrip('$')
+        def _get_template_name(value: str) -> Optional[str]:
+            return value.lstrip(_TEMPLATE_MARKER) if value.startswith(_TEMPLATE_MARKER) else None
 
-        if isinstance(var, Tree) and var.data == 'var' and var.children[0].startswith('$'):
-            return var.children[0].lstrip('$')
+        if isinstance(var, str):

```

Comment:
Incorrect.

If it doesn't start with `$`, it should return None, since it's not a template variable.

But why did you change this code in the first place?

Reply:
`_get_template_name()` will return `None` if the value doesn't start with `$`.

The code had to be updated because the `Tree` portion of this function assumed that the child was a `str`. Since I was touching the code, I opted to remove some of the code duplication.

I just went back to the test cases and noticed that there are [no test cases that cover this file](https://app.codecov.io/gh/lark-parser/lark/blob/master/lark/tree_templates.py). It looks like `py3to2` is the only use of this file in the repository. Based on that, it looks like the annotation for `Template` is not correct (should take `TreeOrCode`). I think I'll take a step back to add test cases, then take a second pass at this.

Reply:
Oh right, my bad.

Anyway, `_get_template_name()` would be better as a global function instead of a closure.

Yes, it's a pretty new feature, not super documented yet either.